core/common/src/main/java/alluxio/wire/BackupStatus.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.wire;
+
+import alluxio.AlluxioURI;
+import alluxio.exception.AlluxioException;
+import alluxio.grpc.BackupPStatus;
+import alluxio.grpc.BackupState;
+
+import com.google.common.base.MoreObjects;
+import com.google.protobuf.ByteString;
+import org.apache.commons.lang.SerializationUtils;
+
+/**
+ * Defines the status of a backup.
+ */
+public class BackupStatus {
```

Comment:
Should this class be immutable?

Reply:
It's used as a wire type. I think it's OK to leave it mutable as with other types. Unless you have a particular reason?

Reply:
Oh I see, it would be nice to make our wire types immutable, but I'm fine with this since its the way we do it for our other ones.